src/riak_kv_qry_compiler.erl

```diff
@@ -1982,10 +2001,110 @@ compile_query_with_arithmetic_type_error_2_test() ->
           "SELECT 2*(location + 1) FROM GeoCheckin "
           "WHERE time > 5000 AND time < 10000"
           "AND user = 'user_1' AND location = 'derby'"),
-    io:format(user, "~p", [compile(get_standard_ddl(), Q, 100)]),
     ?assertEqual(
         {error,{invalid_query,<<"\nOperator '+' called with mismatched types [varchar vs sint64].">>}},
         compile(get_standard_ddl(), Q, 100)
     ).
 
+flexible_keys_1_test() ->
+    DDL = get_ddl(
+        "CREATE TABLE tab4("
+        "a1 SINT64 NOT NULL, "
+        "a TIMESTAMP NOT NULL, "
+        "b VARCHAR NOT NULL, "
+        "c VARCHAR NOT NULL, "
+        "d SINT64 NOT NULL, "
+        "PRIMARY KEY  ((a1, quantum(a, 15, 's')), a1, a, b, c, d))"),
+    {ok, Q} = get_query(
+          "SELECT * FROM tab4 WHERE a > 0 AND a < 1000 AND a1 = 1"),
+    ?assertMatch(
+        {ok, [#riak_select_v1{}]},
+        compile(DDL, Q, 100)
+    ).
+
+quantum_field_name_test() ->
+    DDL = get_ddl(
+        "CREATE TABLE tab1("
+        "a SINT64 NOT NULL, "
+        "b TIMESTAMP NOT NULL, "
+        "PRIMARY KEY  ((a,quantum(b, 15, 's')), a,b))"),
+    ?assertEqual(
+        <<"b">>,
+        quantum_field_name(DDL)
+    ).
+
+quantum_field_name_no_quanta_test() ->
+    DDL = get_ddl(
+        "CREATE TABLE tab1("
+        "a SINT64 NOT NULL, "
+        "b TIMESTAMP NOT NULL, "
+        "PRIMARY KEY  ((a,b), a,b))"),
+    ?assertEqual(
+        no_quanta,
+        quantum_field_name(DDL)
+    ).
+
+%% short key, partition and local keys are the same
+no_quantum_in_query_1_test() ->
```

Comment:
Could you add a variant with a single partition key / local key without a quanta and one with?  If this is covered in a riak test or somewhere else, just say - I don't want to overly duplicate testing effort.